Library/Homebrew/download_strategy.rb

```diff
@@ -312,6 +323,11 @@ def _fetch
     curl @url, "-C", downloaded_size, "-o", temporary_path
   end
 
+  def actual_url(url)
+    m = Utils.popen_read("curl", "-I", "-L", url).match /^Location: (.+)$/
```

Comment:
I'm having an error `bad URI(is not URI?): http://sourceforge.net/projects/upx/files/upx/3.91/upx-3.91-src.tar.bz2`, though this is most probably due to something I hosed while creating an isolated test snippet.

In the meanwhile, this version works:
```ruby
def actual_url(url, &block)
  Utils.popen_read("curl", "-L", "-I", "-o", "/dev/null", "-w", "%{url_effective}", url)
end
```